File: client/src/enums/appearance/gender.ts
/**
 * Represents gender visualization options for image generation.
 * Used in the User model to store appearance preferences.
 */
export enum Gender {
  MAN = "man",
  WOMAN = "woman",
  PERSON = "person", // Gender neutral
}

/**
 * Maps gender values to their human-readable display names
 */
export const GENDER_DISPLAY_NAMES: Record<Gender, string> = {
  [Gender.MAN]: "Man",
  [Gender.WOMAN]: "Woman",
  [Gender.PERSON]: "Person",
};

File: server/enums/appearance/gender.py
from django.db import models


class Gender(models.TextChoices):
    """
    Enum for gender visualization options in image generation.
    """

    MAN = "man", "Man"
    WOMAN = "woman", "Woman"
    PERSON = "person", "Person"  # Gender neutral

    def __str__(self) -> str:
        """
        Get a human-readable string representation of the gender.
        """
        return self.label
